crypto/asn1/a_mbstr.c

```diff
@@ -272,12 +270,15 @@ static int out_utf8(unsigned long value, void *arg)
 static int type_str(unsigned long value, void *arg)
 {
     unsigned long types;
+    const int native = ossl_fromascii(value);
```

Comment:
From the [standard](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) section 6.3.1.3:

> _When a value with integer type is converted to another integer type other than _Bool, if
the value can be represented by the new type, it is unchanged._
> ...
> _Otherwise, the new type is signed and the value cannot be represented in it; either the result is implementation-defined or an implementation-defined signal is raised._

This applies to both implicit and explicit casts.  I don't think there is a formally correct way to do this even if we switch to _long_.

Even more interesting, I can't find a statement that an _int_ and an _unsigned_ have the same number of bits, the _int_ seems to be allowed to be smaller but not larger (6.2.6.2)!